### PR TITLE
Jaro winkler

### PR DESCRIPTION
**Pull request by mateuszz0000**
### **Describe your change:**
Added Jaro Winkler algorithm for measuring similarity between two sequences.
https://en.wikipedia.org/wiki/Jaro%E2%80%93Winkler_distance

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: strings/jaro_winkler.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+""" https://en.wikipedia.org/wiki/Jaro%E2%80%93Winkler_distance """
+import math
+
+
+def jaro_winkler(first_string: str, second_string: str) -> float:
```
Comment (cclauss) [inline strings/jaro_winkler.py:5]:
```suggestion
def jaro_winkler(str1: str, str2: str) -> float:
```
Might make the code easier to read.

Comment (mateuszz0000) [reply strings/jaro_winkler.py]: Fixed. Also found bug in implementation so fixed too.

File: strings/jaro_winkler.py [hunk 2]
```diff
@@ -20,36 +20,50 @@ def jaro_winkler(first_string: str, second_string: str) -> float:
...
+    0.4666666666666666
+
+    >>> jaro_winkler("hell**o", "*world")
+    0.4365079365079365
     """
-    max_distance = math.floor(max(len(first_string), len(second_string)) / 2) - 1
+
+    def get_matched_characters(_str1, _str2):
+        matched = []
+        limit = math.floor(min(len(_str1), len(_str2)) / 2)
```
Comment (cclauss) [inline strings/jaro_winkler.py:37]:
```suggestion
        limit = min(len(_str1), len(_str2)) // 2
```

File: strings/jaro_winkler.py [hunk 3]
```diff
@@ -20,36 +20,50 @@ def jaro_winkler(first_string: str, second_string: str) -> float:
...
+        for i, l in enumerate(_str1):
+            left, right = int(max(0, i - limit)), int(min(i + limit + 1, len(_str2)))
```
Comment (cclauss) [inline strings/jaro_winkler.py:39]:
```suggestion
            left = int(max(0, i - limit))
            right = int(min(i + limit + 1, len(_str2)))
```

File: strings/jaro_winkler.py [hunk 4]
```diff
@@ -20,36 +20,50 @@ def jaro_winkler(first_string: str, second_string: str) -> float:
...
+            if l in _str2[left:right]:
+                matched.append(l)
+                _str2 = _str2[0:_str2.index(l)] + " " + _str2[_str2.index(l) + 1:]
```
Comment (cclauss) [inline strings/jaro_winkler.py:42]:
```suggestion
                _str2 = f"{_str2[0:_str2.index(l)]} {_str2[_str2.index(l) + 1:]}"
```

**Comment by cclauss** [comment]
Please note the formatting changes made by psf/black https://github.com/TheAlgorithms/Python/pull/2051/commits/32109bdcfd52f87d99d0c6950ffc699e17b97854

**Pull request merged**